src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,347 @@
+import classNames from 'classnames';
+import { NavLink, useParams } from 'react-router-dom';
+import { client } from '../../utils/fetchClient';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { PeopleFilters } from '../PeopleFilters';
+import { useTableContext } from '../Context';
+import { SearchLink } from '../SearchLink/SearchLink';
+
+export const PeoplePage: React.FC = () => {
+  const {
+    peoples,
+    setPeoples,
+    hasError,
+    setHasError,
+    isLoading,
+    setIsLoading,
+    necessaryPeople,
+    query,
+    gender,
+    centuries,
+    setCurrPeoples,
+    sortCurr,
+    order,
+    searchParams,
+  } = useTableContext();
+
+  const { slug } = useParams();
+
+  const sortParams = (sortType: string) => {
+    if (sortCurr !== sortType) {
+      return { sort: sortType, order: null };
+    }
+
+    if (order) {
+      return { sort: null, order: null };
+    }
+
+    return { sort: sortType, order: 'desc' };
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    client.get('people.json')
+      .then((response: unknown) => {
+        setPeoples(response as Person[]);
+      })
+      .catch(() => {
+        setHasError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [setHasError, setIsLoading, setPeoples]);
+
+  const visiblePeoples = useMemo(() => {
+    let filteredPeople: Person[] = JSON
+      .parse(JSON.stringify(peoples as Person[]));
+
+    if (query !== '') {
+      filteredPeople = filteredPeople && filteredPeople.filter(person => {
+        const normalizeQuery = query && query.toLowerCase();
+        const normalizeName = person.name.toLowerCase();
+
+        return (
+          normalizeName.includes(normalizeQuery)
+          || (person.fatherName && person.fatherName
+            .toLowerCase().includes(normalizeQuery))
+          || (person.motherName && person.motherName
+            .toLowerCase().includes(normalizeQuery))
+        );
+      });
+    }
+
+    if (gender) {
+      filteredPeople = filteredPeople.filter(person => person.sex === gender);
+    }
+
+    if (centuries.length) {
+      filteredPeople = filteredPeople
+        .filter(person => {
+          const century = Math.ceil(person.died / 100);
+
+          if (centuries.includes(century.toString())) {
+            return person;
+          }
+
+          return null;
+        });
+    }
+
+    switch (sortCurr) {
+      case 'name':
+      case 'sex':
+        filteredPeople.sort((a, b) => (a[sortCurr].localeCompare(b[sortCurr])));
+        break;
+
+      case 'born':
+      case 'died':
+        filteredPeople.sort((a, b) => (a[sortCurr] - b[sortCurr]));
+        break;
+
+      default:
+        break;
+    }
+
+    if (order) {
+      filteredPeople.reverse();
+    }
+
+    return filteredPeople;
+  }, [searchParams, peoples]);
+
+  useEffect(() => {
+    setCurrPeoples(visiblePeoples);
+  }, [peoples, searchParams]);
+
```

Comment:
![2024-01-03_16h24_31](https://github.com/mate-academy/react_people-table-advanced/assets/61904995/8dd012be-dd30-484e-9d3a-99bcb9e04f0f)

Reply:
if you fix these comments, tests freeze
![Screenshot 2024-01-07 170607](https://github.com/mate-academy/react_people-table-advanced/assets/95129845/f7a8c634-d3fc-40ba-8798-0d7eb25210d0)